File: app/src/main/java/fr/erictruong/material/training/ui/components/lists_controls/ListsControlsFragment.java
package fr.erictruong.material.training.ui.components.lists_controls;

import android.os.Bundle;
import android.view.View;

import java.util.ArrayList;
import java.util.List;

import fr.erictruong.android.cards.DisplayBodyItem;
import fr.erictruong.android.cards.HeadlineBodyItem;
import fr.erictruong.android.cards.MaterialCardItem;
import fr.erictruong.material.training.R;
import fr.erictruong.material.training.ui.core.BaseCardFragment;

import static fr.erictruong.android.cards.MaterialCardAdapter.VIEW_TYPE_DISPLAY1_PRIMARY_BODY2;
import static fr.erictruong.android.cards.MaterialCardAdapter.VIEW_TYPE_HEADLINE_PRIMARY_BODY1_THREE_BUTTON;
import static fr.erictruong.android.lists.MaterialListAdapter.VIEW_TYPE_ONE_LINE_AVATAR_CHECKBOX;
import static fr.erictruong.android.lists.MaterialListAdapter.VIEW_TYPE_ONE_LINE_CHECKBOX_ICON;
import static fr.erictruong.android.lists.MaterialListAdapter.VIEW_TYPE_ONE_LINE_ICON_EXPAND;
import static fr.erictruong.android.lists.MaterialListAdapter.VIEW_TYPE_ONE_LINE_ICON_SWITCH;

public class ListsControlsFragment extends BaseCardFragment {

    public static ListsControlsFragment newInstance() {
        Bundle args = new Bundle();
        ListsControlsFragment fragment = new ListsControlsFragment();
        fragment.setArguments(args);
        return fragment;
    }

    @Override
    public void onViewCreated(View view, Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);

        List<MaterialCardItem> cards = new ArrayList<>();
        cards.add(new DisplayBodyItem.Builder()
                .viewType(VIEW_TYPE_DISPLAY1_PRIMARY_BODY2)
                .display(getString(R.string.lists_controls))
                .body(getString(R.string.lists_controls_txt))
                .build());
        cards.add(new HeadlineBodyItem.Builder()
                .viewType(VIEW_TYPE_HEADLINE_PRIMARY_BODY1_THREE_BUTTON)
                .headline("Checkbox")
                .button1("Sample", new View.OnClickListener() {
                    @Override
                    public void onClick(View v) {
                        SampleListControlActivity.start(getContext(), VIEW_TYPE_ONE_LINE_CHECKBOX_ICON);
                    }
                })
                .button2("Alt", new View.OnClickListener() {
                    @Override
                    public void onClick(View v) {
                        SampleListControlActivity.start(getContext(), VIEW_TYPE_ONE_LINE_AVATAR_CHECKBOX);
                    }
                })
                .build());
        cards.add(new HeadlineBodyItem.Builder()
                .viewType(VIEW_TYPE_HEADLINE_PRIMARY_BODY1_THREE_BUTTON)
                .headline("Switch")
                .button1("Sample", new View.OnClickListener() {
                    @Override
                    public void onClick(View v) {
                        SampleListControlActivity.start(getContext(), VIEW_TYPE_ONE_LINE_ICON_SWITCH);
                    }
                })
                .build());
        cards.add(new HeadlineBodyItem.Builder()
                .viewType(VIEW_TYPE_HEADLINE_PRIMARY_BODY1_THREE_BUTTON)
                .headline("Reorder")
                .button1("Sample", null)
                .build());
        cards.add(new HeadlineBodyItem.Builder()
                .viewType(VIEW_TYPE_HEADLINE_PRIMARY_BODY1_THREE_BUTTON)
                .headline("Switch")
                .button1("Sample", new View.OnClickListener() {
                    @Override
                    public void onClick(View v) {
                        SampleListControlActivity.start(getContext(), VIEW_TYPE_ONE_LINE_ICON_EXPAND);
                    }
                })
                .build());
        cards.add(new HeadlineBodyItem.Builder()
                .viewType(VIEW_TYPE_HEADLINE_PRIMARY_BODY1_THREE_BUTTON)
                .headline("Leave-behinds")
                .button1("Sample", null)
                .build());

        getAdapter().swapData(cards);
    }
}

File: app/src/main/java/fr/erictruong/material/training/ui/components/lists_controls/ListsControlsFragment.kt
package fr.erictruong.material.training.ui.components.lists_controls

import android.os.Bundle
import android.view.View
import fr.erictruong.android.cards.item.DisplayBodyItem
import fr.erictruong.android.cards.item.HeadlineBodyThreeButtonItem
import fr.erictruong.android.cards.item.MaterialCardItem
import fr.erictruong.android.core.util.ThemeUtils
import fr.erictruong.material.training.R
import fr.erictruong.material.training.ui.core.BaseCardFragment
import org.jetbrains.anko.startActivity
import org.jetbrains.anko.support.v4.ctx
import java.util.*

class ListsControlsFragment : BaseCardFragment() {

    override fun onViewCreated(view: View?, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        val colorPrimary = ThemeUtils.obtainColorPrimary(ctx)

        val cards = ArrayList<MaterialCardItem>()
        cards.add(DisplayBodyItem(
                display = getString(R.string.lists_controls), displayColor = colorPrimary,
                body = getString(R.string.lists_controls_txt), bodyAppearance = R.style.TextAppearance_AppCompat_Body2))
        cards.add(HeadlineBodyThreeButtonItem(
                headline = "Checkbox", headlineColor = colorPrimary,
                button1 = "Sample", button1Action = { ctx.startActivity<CheckboxIconListControlActivity>() },
                button2 = "Alt", button2Action = { ctx.startActivity<AvatarCheckboxListControlActivity>() }))
        cards.add(HeadlineBodyThreeButtonItem(
                headline = "Switch", headlineColor = colorPrimary,
                button1 = "Sample", button1Action = { ctx.startActivity<IconSwitchListControlActivity>() }))
        cards.add(HeadlineBodyThreeButtonItem(
                headline = "Reorder", headlineColor = colorPrimary,
                button1 = "Sample", button1Action = { ctx.startActivity<AvatarReorderListControlActivity>() }))
        cards.add(HeadlineBodyThreeButtonItem(
                headline = "Expand/collapse", headlineColor = colorPrimary,
                button1 = "Sample", button1Action = { ctx.startActivity<IconExpandListControlActivity>() }))
        cards.add(HeadlineBodyThreeButtonItem(
                headline = "Leave-behinds", headlineColor = colorPrimary,
                button1 = "Sample", button1Action = null)) // TODO: Write and start LeaveBehindsListControlActivity.

        adapter.swapData(cards)
    }
}
